code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -2472,6 +2472,56 @@
 		T.health -= 20
 		T.mutation_mod += 0.1
 
+/datum/reagent/toxin/insecticide
+	name = "Insecticide"
+	id = INSECTICIDE
+	description = "A broad pesticide. Do not ingest!"
+	reagent_state = REAGENT_STATE_LIQUID
+	color = "#49002E" //rgb: 73, 0, 46
+	density = 1.08
+	specheatcap = 4.18
+
+/datum/reagent/toxin/insecticide/reaction_mob(var/mob/living/M, var/method = TOUCH, var/volume)
+
+	if(..())
+		return 1
+	if(iscarbon(M))
+		var/mob/living/carbon/C = M
+		if(!C.wear_mask) //If not wearing a mask
+			C.adjustToxLoss(REM) //4 toxic damage per application, doubled for some reason
+		if(isinsectoid(C) || istype(C, /mob/living/carbon/monkey/roach)) //Insecticide being poisonous to bugmen, who'd've thunk
+			M.adjustToxLoss(10 * REM)
+					
+/datum/reagent/toxin/insecticide/reaction_animal(var/mob/living/simple_animal/M, var/method=TOUCH, var/volume)
+	if(..())
+		return 1
+		
+	if(istype(M,/mob/living/simple_animal/hostile/bigroach))
+		M.gib()
+		
+	if(istype(M,/mob/living/simple_animal/cockroach))
+		M.gib()
+	
+	if(istype(M,/mob/living/simple_animal/bee))
+		M.gib()
+
+	if(istype(M,/mob/living/simple_animal/hostile/giant_spider))
+		M.gib()
+	
+/datum/reagent/toxin/insecticide/reaction_obj(var/obj/O, var/volume)
+
+	if(..())
+		return 1
+	if(istype(O,/obj/machinery/portable_atmospherics/hydroponics))
+		var/obj/machinery/portable_atmospherics/hydroponics/tray = O
+		tray.pestlevel -= 2
+		
+/datum/reagent/toxin/insecticide/on_plant_life(obj/machinery/portable_atmospherics/hydroponics/T)
+	..()
```

Comment:
This isn't your fault but I notice the parent proc of this does not work as it should; it handles the sanity but those return 0, meaning you can't add an if statement here to avoid runtimes if it has no holder somehow.

You don't need to change this though, since it seems very unlikely to happen. Just pointing it out for general notice.

Reply:
GOD I wish I knew what that meant codewise.

Reply:
Don't worry about this since it's above your level

Reply:
welp